pkg/privilege/privileges/privileges.go

```diff
@@ -930,6 +930,79 @@ func (p *UserPrivileges) ShowGrants(ctx sessionctx.Context, user *auth.UserIdent
 	return
 }
 
+// FetchColumnPrivileges implements privilege.Manager interface.
+func (p *UserPrivileges) FetchColumnPrivileges(sctx sessionctx.Context, user *auth.UserIdentity) ([][]types.Datum, error) {
+	if SkipWithGrant {
+		return nil, ErrNonexistingGrant.GenWithStackByArgs("root", "%")
+	}
+
+	mysqlPrivilege := p.Handle.Get()
+	users := make([]string, 0, 1)
+	hosts := make([]string, 0, 1)
+
+	u, h := auth.GetUserAndHostName(user)
+	if u == "" || p.RequestVerification(sctx.GetSessionVars().ActiveRoles, "mysql", "columns_priv", "", mysql.SelectPriv) &&
```

Comment:
What's the behavior considering RBAC, should the user's role's column privileges be included?

Reply:
In mysql's behavior, role affects the result of `show grants`, but not affects these three information_schema views.